packages/drivers/odsp-driver/src/odspDriverUrlResolver.ts

```diff
@@ -140,7 +140,7 @@ export class OdspDriverUrlResolver implements IUrlResolver {
         }
         const odspResolvedUrl = resolvedUrl as IOdspResolvedUrl;
         let odspUrl = `${odspResolvedUrl.siteUrl}/${url}?driveId=${encodeURIComponent(odspResolvedUrl.driveId,
-            )}&itemId=${encodeURIComponent(odspResolvedUrl.itemId)}&path=${encodeURIComponent("/")}`;
+            )}&itemId=${encodeURIComponent(odspResolvedUrl.itemId)}&path=${encodeURIComponent(url)}`;
```

Comment:
`url` is encoded twice in odspUrl now. What is `path` supposed to have?

Reply:
Path is the data store path.

Reply:
It can be "any" path that needs to be passed to the Container right; doesn't have to be a data store path. I am a little concerned that it is now part of the osdpUrl twice. Can we see a real example of what this URL looks like for a fluid document?

Reply:
Also, we should add tests that covers the changes here.

Reply:
Sorry what other path would it be if not the data store path?

Reply:
It can be anything that the app wants to pass to the container. You can register request handlers to process any request. For example, path to a blob, initial configs, etc.

Reply:
@wes-carlson So url is already there in odspurl, why do we need to add it twice?

Reply:
I originally tried changing `resolve()` to consume the path from the url pathname where `getAbsoluteUrl()` puts it currently, but this caused other unit tests to fail:
```
fluid url does not match

      + expected - actual

      -fluid-odsp://placeholder/placeholder/iNKF5NbXkQ9HFHj62PjIkOuQxDamm0zbffgsz8gigBA%3D/_api/v2.1/drives/randomDrive/items/randomItem/?driveId=randomDrive&itemId=randomItem&path=
      +fluid-odsp://placeholder/placeholder/iNKF5NbXkQ9HFHj62PjIkOuQxDamm0zbffgsz8gigBA%3D/?driveId=randomDrive&itemId=randomItem&path=
```

These seem more like "real" test failures than the ones that are currently failing, since I don't think `_api/v2.1/drives/randomDrive/items/randomItem` should be in the url.

Reply:
Yes instead of "_api/v2.1/drives/randomDrive/items/randomItem", I think it should be the path?

Reply:
I think we should discuss with ODSP folks as to what the expected behavior here is. If we return an incorrect url here, everything will fail!

Reply:
@wes-carlson We are already adding the relative path passed in getAbsoluteUrl api to the returned URL.  Does it create wrong urls where we are not able to correctly route the request to apt datastore?

Reply:
So I think the relative path that we are passing to getAbsoluteUrl is wrong, as per your example?

Reply:
> Does it create wrong urls where we are not able to correctly route the request to apt datastore?

Yes, there's more detail in the linked issue (#5359), but `resolve()` puts the pathname of the url into `siteUrl` but not `url`, which is used by loader.

> So I think the relative path that we are passing to getAbsoluteUrl is wrong, as per your example?

Not sure what you mean, where are we doing that?